r-uefi-pi/src/hob.rs

```diff
@@ -0,0 +1,260 @@
+// Copyright © 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::boot_mode::BootMode;
+use r_efi::efi::{Guid, PhysicalAddress};
+use scroll::{Pread, Pwrite};
+
+pub const HOB_LIST_GUID: Guid = Guid::from_fields(
+    0x7739F24C,
+    0x93D7,
+    0x11D4,
+    0x9A,
+    0x3A,
+    &[0x00, 0x90, 0x27, 0x3F, 0xC1, 0x4D],
+);
+
+pub type ResourceType = u32;
+
+pub const RESOURCE_SYSTEM_MEMORY: u32 = 0x00;
+pub const RESOURCE_MEMORY_MAPPED_IO: u32 = 0x01;
+pub const RESOURCE_IO: u32 = 0x02;
+pub const RESOURCE_FIRMWARE_DEVICE: u32 = 0x03;
+pub const RESOURCE_MEMORY_MAPPED_IO_PORT: u32 = 0x04;
+pub const RESOURCE_MEMORY_RESERVED: u32 = 0x05;
+pub const RESOURCE_IO_RESERVED: u32 = 0x06;
```

Comment:
It would be great to convert these constants into an enum.

Reply:
this matches C definition.
I am not sure if we need rust enum type.